libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -267,7 +276,11 @@ public void userOffRoute(Location location) {
     @Override
     public void onMilestoneEvent(RouteProgress routeProgress, String instruction, Milestone milestone) {
       if (milestone instanceof VoiceInstructionMilestone) {
-        instructionPlayer.play((VoiceInstructionMilestone) milestone);
+        SpeechAnnouncement speechAnnouncement = SpeechAnnouncement.builder()
```

Comment:
what if you pass in `VoiceInstructionMilestone` into a constructor in `SpeechAnnouncement` so this logic can be in `SpeechAnnouncement`? Instead of using the builder pattern

Reply:
Was thinking here in terms of usability outside of the turn-by-turn UI.  `NavigationSpeechPlayer` is a `public` API, so tying it completely to the `VoiceInstructionMilestone` in order to voice something may not make a lot of sense to someone that doesn't want to use the milestones to voice something.  

I did update the builder though to also take a `VoiceInstructionMilestone`, does that work?   Any reasons you don't like the builder pattern here?

Reply:
That makes sense! And yea having a builder method that takes in a `VoiceInstructionMilestone` for convenience is 👍